Dockerfile

```diff
@@ -1,30 +1,65 @@
 # syntax=docker/dockerfile:experimental
-FROM golang:1.14.4-alpine3.12 AS build_base
+
+FROM pingcap/chaos-build-base AS go_build
+
+RUN curl https://dl.google.com/go/go1.14.6.linux-amd64.tar.gz | tar -xz -C /usr/local
+ENV PATH "/usr/local/go/bin:${PATH}"
+ENV GO111MODULE=on
 
 ARG HTTPS_PROXY
 ARG HTTP_PROXY
 
-RUN apk add --no-cache gcc g++ make bash git
-RUN apk add --update nodejs yarn
+RUN if [[ -n "$HTTP_PROXY" ]]; then yarn config set proxy $HTTP_PROXY; fi
 
-ENV GO111MODULE=on
 WORKDIR /src
-COPY go.mod .
-COPY go.sum .
-
-RUN go mod download
 
-FROM build_base AS binary_builder
+COPY . /src
 
-ARG HTTPS_PROXY
-ARG HTTP_PROXY
 ARG UI
 ARG SWAGGER
 ARG LDFLAGS
 
-RUN if [[ -n "$HTTP_PROXY" ]]; then yarn config set proxy $HTTP_PROXY; fi
-
-COPY . /src
-WORKDIR /src
-RUN --mount=type=cache,target=/root/.cache/go-build \
+RUN --mount=type=cache,target=/root/go/pkg \
+    --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/src/ui/node_modules \
     IMG_LDFLAGS=$LDFLAGS make binary
+
+FROM pingcap/chaos-build-base AS rust_build
+
+ARG HTTPS_PROXY
+ARG HTTP_PROXY
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain nightly-2020-07-01 -y
```

Comment:
We don't need to build `toda` every time.  Just use `toda` as a single project and use the release image at this repo.

Reply:
I agree with you :+1: